src/components/SQForm/__tests__/SQFormDropdown.test.js

```diff
@@ -0,0 +1,202 @@
+import React from 'react';
+import * as Yup from 'yup';
+import {screen} from '@testing-library/react';
+import SQFormDropdown from '../SQFormDropdown';
+import {renderSQForm} from '../../../utils/tests';
+import userEvent from '@testing-library/user-event';
+
+const DEFAULT_PROPS = {
+  name: 'dropdownExample',
+  label: 'Dropdown Example'
+};
+
+const DEFAULT_INTITIAL_VALUES = {
+  initialValues: {
+    dropdownExample: ''
+  }
+};
+
+const DEFAULT_OPTIONS = [
+  {label: 'Test 1', value: 1},
+  {label: 'Test 2', value: 2},
+  {label: 'Test 3', value: 3}
+];
+
+const createDropdown = (
+  options = DEFAULT_OPTIONS,
+  fieldProps = DEFAULT_PROPS
+) => {
+  return <SQFormDropdown {...fieldProps}>{options}</SQFormDropdown>;
+};
+
+it('should render list of options', () => {
+  const renderDropdown = createDropdown();
+
+  renderSQForm(renderDropdown, DEFAULT_INTITIAL_VALUES);
+
+  const expandButton = screen.getByRole('button', /dropdown example/i);
+  userEvent.click(expandButton);
+
+  const optionsList = screen.getByRole('listbox', /dropdown example/i);
+
+  expect(optionsList).toBeInTheDocument();
+  expect(optionsList.children.length).toBe(3);
+});
+
+it('should render with initial value', () => {
+  const initialValues = {
+    dropdownExample: 2
+  };
+
+  const renderDropdown = createDropdown();
+
+  renderSQForm(renderDropdown, {initialValues});
+
+  const labelValue = screen.getByRole('button', /dropdown example/i);
+  expect(labelValue).toHaveTextContent('Test 2');
+});
+
+it('should update when option is selected', () => {
+  const handleChange = jest.fn();
+  const fieldProps = {
+    ...DEFAULT_PROPS,
+    onChange: handleChange
+  };
+
+  const renderDropdown = createDropdown(DEFAULT_OPTIONS, fieldProps);
+
+  renderSQForm(renderDropdown, DEFAULT_INTITIAL_VALUES);
+
+  const expandButton = screen.getByRole('button', /dropdown example/i);
+  userEvent.click(expandButton);
+
+  const option = screen.getByText('Test 1');
+  userEvent.click(option);
+
+  expect(handleChange).toHaveBeenCalled();
```

Comment:
Do we need to have two expects here.  I think the one that specifies how many times it was called is sufficient. We can remove the one on line 76.

Reply:
> Do we need to have two expects here. I think the one that specifies how many times it was called is sufficient. We can remove the one on line 76.

+1

Question - `expect(handleChange).toHaveBeenCalled();` Reads really well. Are we concerned with how many times it's been called? If there was a bug I guess it guards against that possibility? 

If we have to say `expect(handleChange).toHaveBeenCalledTimes(1);` it's hard for me to picture when `expect(handleChange).toHaveBeenCalled();` would ever be used?

Question 2 - Where do we test what args `handleChange` is called with? Meaning, how do we ensure it's being called with the expected arguments? Or do we care?

Reply:
Personally, I would just test whatever `handleChange` does rather than testing that it was called. If all `handleChange` does is change the selected value then we could just test that and leave out the `toHaveBeenCalled`

Reply:
I do like the `toHaveBeenCalled` with the precision of how many times to cover scenarios where it get's triggered more than once by accident, a bug in the code, or a re-render.

Reply:
> Personally, I would just test whatever `handleChange` does rather than testing that it was called. If all `handleChange` does is change the selected value then we could just test that and leave out the `toHaveBeenCalled`

I think this is what i'm looking for. `toHavebeenCalled` or the number of times it's been called doesn't give me much confidence the function is working properly. 

Testing what the function should do, does give me confidence the function is working as expected.

Reply:
So right under this I have
`const labelValue = screen.getByRole('button', /dropdown example/i);
  expect(labelValue).toHaveTextContent('Test 1');`
  to test that it actually changes the value.

I can remove these expects all together if we think they're not useful.